naga/src/front/spv/image.rs

```diff
@@ -377,36 +377,54 @@ impl<I: Iterator<Item = u32>> super::Frontend<I> {
         let coord_handle =
             self.get_expr_handle(coordinate_id, coord_lexp, ctx, emitter, block, body_idx);
         let coord_type_handle = self.lookup_type.lookup(coord_lexp.type_id)?.handle;
-        let (coordinate, array_index) = match ctx.type_arena[image_ty].inner {
+        let (coordinate, array_index, is_depth) = match ctx.type_arena[image_ty].inner {
             crate::TypeInner::Image {
                 dim,
                 arrayed,
-                class: _,
-            } => extract_image_coordinates(
-                dim,
-                if arrayed {
-                    ExtraCoordinate::ArrayLayer
-                } else {
-                    ExtraCoordinate::Garbage
-                },
-                coord_handle,
-                coord_type_handle,
-                ctx,
-            ),
+                class,
+            } => {
+                let (coord, array_index) = extract_image_coordinates(
+                    dim,
+                    if arrayed {
+                        ExtraCoordinate::ArrayLayer
+                    } else {
+                        ExtraCoordinate::Garbage
+                    },
+                    coord_handle,
+                    coord_type_handle,
+                    ctx,
+                );
+                (coord, array_index, class.is_depth())
+            }
             _ => return Err(Error::InvalidImage(image_ty)),
         };
 
-        let expr = crate::Expression::ImageLoad {
+        let image_load_expr = crate::Expression::ImageLoad {
             image: image_lexp.handle,
             coordinate,
             array_index,
             sample,
             level,
         };
+        let image_load_handle = ctx
+            .expressions
+            .append(image_load_expr, self.span_from_with_op(start));
+
+        let handle = if is_depth {
+            let splat_expr = crate::Expression::Splat {
+                size: crate::VectorSize::Quad,
+                value: image_load_handle,
+            };
```

Comment:
The result type of `OpImageRead` is documented as:

> Result Type must be a scalar or vector of floating-point type or integer type.

So I think we should splat based on the nr of components in the return type.